expo/expo
# [expo-dev-launcher] add updates integration to config plugin

# Why

If you have both expo-updates and expo-dev-client installed in a managed project, we should assume you want the integration to work in your native/bare/prebuilt project.

# How

In the dev-launcher config plugin, check for the existence of the expo-updates package (in the same way as the [react-native-maps plugin](https://github.com/expo/expo-cli/blob/master/packages/config-plugins/src/ios/Maps.ts)) and, if it's installed, make the few additional changes from https://github.com/expo/expo/pull/13197.

# Test Plan

Added a snapshot test to go alongside the existing one for iOS, and tested manually using the following steps:
- expo init --> blank template
- `yarn add expo-dev-client expo-updates@next`
- copy local expo-dev-launcher plugin into node_modules
- `expo prebuild --no-install --skip-dependency-update expo-updates`
- check MainApplication.java and AppDelegate.m to make sure they have the correct changes
- yarn & pod install, build projects, load development and production projects in dev client

To test the case without updates, follow the above steps but install `expo-updates@latest` and just run `expo prebuild --no-install` (no `--skip-dependency-update`).

# Checklist
